### PR TITLE
EPMLSTRCMW-166: Implement delete endpoint for project confguration

### PR DESCRIPTION
**Pull request by Dmitrii-Shevchenko**
[PR](https://github.com/EpamLifeSciencesTeam/cmwl_pipeline/pull/84)

### CODE REVIEW

File: repositories/src/main/scala/cromwell/pipeline/datastorage/dto/TypedValue.scala [hunk 1]
```diff
@@ -23,7 +23,7 @@ object TypedValue {
     implicit val floatTyped: OFormat[FloatTyped] = Json.format
     implicit val booleanTyped: OFormat[BooleanTyped] = Json.format
 
-    adtFormat("$type")(
+    adtFormat("_type")(
```
Comment (OstapStrashevskii) [inline repositories/src/main/scala/cromwell/pipeline/datastorage/dto/TypedValue.scala:26]: what do this mean? could you explain? any regexp?

Comment (Dmitrii-Shevchenko) [reply repositories/src/main/scala/cromwell/pipeline/datastorage/dto/TypedValue.scala:26]: Of course, you can look at conversations on https://github.com/EpamLifeSciencesTeam/cmwl_pipeline/pull/84#discussion_r469159503

File: services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
+      case Some(conf) if conf.isActive =>
+        Some(ProjectConfigurationEntity(conf.projectId, conf.projectFileConfigurations))
+      case _ => None
+    }
+
+  def deactivateConfiguration(projectId: ProjectId): Future[String] =
+    getById(projectId).flatMap {
+      case Some(_) =>
+        repository.updateOneField("projectId", projectId.value, "isActive", false).map(_.toString)
+      case _ => Future.successful("There is no project to deactivate")
```
Comment (iliaonishchenko) [inline services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala]: From what I see, it's not successful path.
If there is no project to deactivate we should return error. But now here and in the controller it's a 200 status.

File: services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala [hunk 3]
```diff
@@ -1,18 +1,34 @@
...
+        toDocument(ProjectConfiguration(configuration)),
+        "projectId",
+        configuration.projectId.value
+      )
       .map(_.toString)
 
-  def getById(projectId: ProjectId): Future[Option[ProjectConfiguration]] =
-    repository.getByParam("projectId", projectId.value).map(_.headOption.map(fromDocument))
+  def getById(projectId: ProjectId): Future[Option[ProjectConfigurationEntity]] =
+    repository.getByParam("projectId", projectId.value).map(_.headOption.map(fromDocument)).map {
```
Comment (iliaonishchenko) [inline services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala]: If we're waiting for 1 or 0 projects, the method should return option, not a sequence.

**Review by iliaonishchenko — CHANGES_REQUESTED** [review]
There are some comments:

File: services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala (hunk 2 above)
Comment (Dmitrii-Shevchenko) [reply services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala]: Done

File: services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala (hunk 3 above)
Comment (Dmitrii-Shevchenko) [reply services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala]: Done

File: services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala [hunk 4]
```diff
@@ -1,18 +1,35 @@
...
+        configuration.projectId.value
+      )
       .map(_.toString)
 
-  def getById(projectId: ProjectId): Future[Option[ProjectConfiguration]] =
-    repository.getByParam("projectId", projectId.value).map(_.headOption.map(fromDocument))
+  def getById(projectId: ProjectId): Future[Option[ProjectConfigurationEntity]] =
+    repository.getByParam("projectId", projectId.value).map { optDocument =>
+      optDocument.filter(document => fromDocument(document).isActive).map { document =>
+        val config = fromDocument(document)
```
Comment (myazinn) [inline services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala:24]: Avoid multiple `fromDocument` parsing.
`optDocument.map(fromDocument).filter(_.isActive).map(config => ???)`

File: services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala [hunk 5]
```diff
@@ -1,18 +1,35 @@
...
+        ProjectConfigurationEntity(config.projectId, config.projectFileConfigurations)
+      }
+    }
+
+  def deactivateConfiguration(projectId: ProjectId): Future[String] =
+    getById(projectId).flatMap {
+      case Some(_) =>
+        repository.updateOneField("projectId", projectId.value, "isActive", false).map(_.toString)
```
Comment (myazinn) [inline services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala:32]: This `.toString` is a real headache. I guess I should create another task to get rid of it.

**Review by myazinn — COMMENTED** [review]
Please fix a small issue and I'll this PR again

**Comment by iliaonishchenko** [comment]
This MR seams to be dead. I'm closing it.

**Pull request closed without merging**